electron_app/electron.js

```diff
@@ -359,6 +373,19 @@ Main._createMainWindow = function () {
     });
   }
 
+  function setSaveSingleDiagram() {
+    electron.ipcMain.on('save_single_diagram', (event) => {
```

Comment:
We use kebab case not snake case for events

```suggestion
    electron.ipcMain.on('save-single-diagram', (event) => {
```